src/legacy/core_plugins/metrics/index.ts

```diff
@@ -45,9 +44,8 @@ export default function(kibana: any) {
 
     init(server: Legacy.Server) {
       const initializerContext = {} as PluginInitializerContext;
-      const core = { http: { server } } as InternalCoreSetup;
 
-      metricsPlugin(initializerContext).setup(core);
+      metricsPlugin(initializerContext).setup(server);
```

Comment:
Sorry, I realize now my recommendation on this earlier was a bit vague: I would still pass the entirety of `core` to `setup` (that's how it will work in the new platform... `core` will be the first argument to `setup`). But then inside of your `plugin.ts`, destructure `const { http } = core` and pass `http.server` to the functions that need it. Apologies for the confusion.

Reply:
Renamed